### PR TITLE
Dark mode improvements

### PR DESCRIPTION
**Pull request by codecalm**
demo: https://deploy-preview-950--tabler.netlify.app/?theme=dark

localhost_3000_index html
localhost_3000_wizard html
localhost_3000_layout-vertical-transparent html
localhost_3000_modals html
localhost_3000_search-results html
screencapture-localhost-3000-modals-html-2021-12-03-00_12_23
localhost_3000_music html
localhost_3000_charts html
localhost_3000_form-elements html
localhost_3000_gallery html

### CODE REVIEW

**Comment by codecalm** [comment]
from now, we can enable dark mode in navbar:

Screenshot 2021-12-03 at 00 07 17

**Comment by kevinpapst** [comment]
That is a great addition, finally no need to use the developer console to change the theme on each demo page :)

Can you share the why you changed from `class="theme-dark"` to `body[data-theme="dark"]`? I don't fully understand the change, but it feels like a big BC break and (without known any background about this change) feels wrong to me. Is this only for the demo? Why did you choose to use a data-attribute instead of a class name? I see the code being used in the demo JS but also some of the the SASS files.

Some points I found: 
- `<pre>` has still white background: https://deploy-preview-950--tabler.netlify.app/docs/form-selectboxes.html
- Same page "Advanced selectboxes": the selected title gets dark and is not visible
- the [highlighted row](https://deploy-preview-950--tabler.netlify.app/tables.html) in `table-striped` should probably blend in more instead of being plain white
- Is it desirable that a [dark button](https://deploy-preview-950--tabler.netlify.app/playground-dark.html?theme=dark) won't be shown in dark theme?

At least the demo has huge differences between modals in the [documentation](https://deploy-preview-950--tabler.netlify.app/docs/modals.html) and on the [modals page](https://deploy-preview-950--tabler.netlify.app/modals.html). Some elements are not shown. It seems the same happens to other elements (see "Advanced selectboxes" above).

Maybe you can have a look at https://github.com/tabler/tabler/issues/945 and include the fixes?

**Comment by rjd22** [comment]
@codecalm looks good to me for a initial version. A few recommendations:

* I would normally use cookies for storing dark mode. This way backend systems can also know if dark mode is enabled
* Is there a technical reason you switched from the class to a data attribute? I prefer a class because it's way simpler to toggle.

I actually made a similar implementation in vue that toggles the dark mode without refreshing the whole screen. This way forms are remembered. I will share it as inspiration and remove the vuejs bits:

```js
    let darkModeEnabled = false;

    const enableDarkMode = (isEnabled) => {
      darkModeEnabled = isEnabled;

      document.body.classList.toggle('theme-dark');
      document.cookie = `darkMode=${isEnabled}; path=/; SameSite=Strict; Secure`;
    };

    const cookies = document.cookie.split(';');
    const userPrefersDarkMode = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;

    if (cookies.includes('darkMode=true') || cookies.includes('darkMode=false')) {
      darkModeEnabled = cookies.includes('darkMode=true');
    } else if (userPrefersDarkMode) {
      enableDarkMode(true);
    }

    return {
      darkModeEnabled,
      enableDarkMode,
    };
```

File: src/pages/_layouts/base.html [hunk 1]
```diff
@@ -39,7 +39,7 @@
 </head>
 
 {% assign layout-dark = page.layout-dark | default: site.layout-dark %}
-<body class="{% if layout-dark %} theme-dark{% endif %}{% if layout.body-class %} {{ layout.body-class }}{% endif %}{% if page.body-class %} {{ page.body-class }}{% endif %}">
+<body {% if layout.body-class or page.body-class %} class="{% if layout.body-class %} {{ layout.body-class }}{% endif %}{% if page.body-class %} {{ page.body-class }}{% endif %}"{% endif %} data-theme="light" data-menu-position="top" data-menu-behavior="fixed" data-container-layout="boxed">
```
Comment (rjd22) [inline src/pages/_layouts/base.html:42]: This is the only thing I'm unsure about. Is there a technical or another reason why classes would not work? I would say that a class is the right use for dark theme's.

**Review by rjd22 — CHANGES_REQUESTED** [review]

File: src/pages/_layouts/base.html (hunk 1 above)
Comment (codecalm) [reply src/pages/_layouts/base.html:42]: no problem, i'll change it

Comment (codecalm) [reply src/pages/_layouts/base.html]: fixed

**Comment by codecalm** [comment]
@rjd22 its just for demo, not light/dark mode mechanism. `demo.js` will not be included in main package.

**Comment by rjd22** [comment]
@codecalm lgtm. :shipit:

**Review by rjd22 — APPROVED** [review]

**Comment by kevinpapst** [comment]
I would ask for the following fix, even if it wasn't part of the original PR intention:
- the highlighted row in table-striped should probably blend in more instead of being plain white

Some content (here: "Egan Poetz") is only visible, if you mark some text with the cursor: 
Bildschirmfoto 2021-12-03 um 15 05 56

Everything else looks great 👍

**Comment by codecalm** [comment]
@kevinpapst fixed :)

**Comment by kevinpapst** [comment]
Found one more: 
divs with `bg-light` as in [Borders](https://deploy-preview-950--tabler.netlify.app/docs/borders.html) and [Cursors](https://deploy-preview-950--tabler.netlify.app/docs/cursors.html) look weird. At least the font color should be adjusted. Or is that only an example content problem?

**Comment by rjd22** [comment]
@codecalm I pushed a separate branch with chart fixes for dark mode to avoid conflicts. If you like you can cherry pick the change:

https://github.com/tabler/tabler/pull/952

**Comment by kevinpapst** [comment]
A tiny one: I believe that `.nav-tabs` currently has no dark styling. 
https://deploy-preview-950--tabler.netlify.app/tabs.html
Seems that these borders are the only ones which are so light.

Seems this was mentioned in #897 before.

**Comment by jaikt** [comment]
som how'd I implement the dark mode and light mode toggle and everything.?

**Pull request merged**